src/utils/polygon.cpp

```diff
@@ -156,6 +156,92 @@ Polygons PolygonRef::offset(int distance, ClipperLib::JoinType joinType, double
     return ret;
 }
 
+Polygon Polygons::convexHull() const
+{
+    // Implements Andrew's monotone chain convex hull algorithm
+    // See https://en.wikibooks.org/wiki/Algorithm_Implementation/Geometry/Convex_hull/Monotone_chain
+
+    using IntPoint = ClipperLib::IntPoint;
+
+    size_t num_points = 0U;
+    for (const ClipperLib::Path &path : paths)
+    {
+        num_points += path.size();
+    }
+
+    std::vector<IntPoint> all_points;
+    all_points.reserve(num_points);
```

Comment:
all_points is just the points from all the polygons concatenated, so it is always exactly num_points in size.  It is only there because we need to sort the points from the original polygons.  The hull_points needs to be of size "num_points + 1" and it is.